mitosheet/src/components/footer/GraphSheetTabActions.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Mito
+
+import React, { useEffect } from 'react';
+import MitoAPI, { getRandomId } from '../../api';
+import { GraphDataJSON, GraphID, UIState } from '../../types';
+import Dropdown from '../elements/Dropdown';
+import DropdownItem from '../elements/DropdownItem';
+import { TaskpaneType } from '../taskpanes/taskpanes';
+
+/*
+    Displays a set of actions one can perform on a graph sheet tab, including
+    deleting, duplicating, or renaming.
+*/
+export default function GraphSheetTabActions(props: {
+    setDisplayActions: React.Dispatch<React.SetStateAction<boolean>>,
+    setIsRename: React.Dispatch<React.SetStateAction<boolean>>;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    closeOpenEditingPopups: () => void;
+    mitoAPI: MitoAPI,
+    graphID: GraphID,
+    graphDataJSON: GraphDataJSON
+}): JSX.Element {
+
+    // Log opening the graph sheet tab actions
+    useEffect(() => {
+        void props.mitoAPI.log(
+            'clicked_graph_sheet_tab_actions',
+            {
+                graph_id: props.graphID
+            }
+        )
+    }, [])
+
+    const onDelete = async (): Promise<void> => {
+        // Close 
+        props.closeOpenEditingPopups();
+
+        await props.mitoAPI.editGraphDelete(props.graphID)
+    }
+
+    const onDuplicate = async (): Promise<void> => {
+        // Close 
+        props.closeOpenEditingPopups();
+        
+        // Create the newGraphID so we can select the new graph tab
+        const newGraphID = getRandomId()
+        await props.mitoAPI.editGraphDuplicate(props.graphID, newGraphID)
+
+        props.setUIState(prevUIState => {
+            return {
+                ...prevUIState,
+                selectedGraphID: newGraphID,
+                selectedTabType: 'graph',
+                currOpenTaskpane: {
+                    type: TaskpaneType.GRAPH,
+                    graphID: newGraphID,
+                },
+            }
+        })
```

Comment:
I think we can (and should) remove this -- this is now handled by the effect that watches for new graphs, right? Let's not duplicate the behavior here for no reason.